pmd-core/src/main/java/net/sourceforge/pmd/ant/Formatter.java

```diff
@@ -57,12 +61,26 @@ public Renderer getRenderer() {
     }
 
     public void start(String baseDir) {
+
+        Properties properties = createProperties();
+        
+        Charset charset;
+        {
+            String s = (String) properties.get("encoding");
+            if (null == s) {
+                charset = StandardCharsets.UTF_8;
+                properties.put("encoding", charset.name());
+            } else {
+                charset = Charset.forName(s);
+            }
+        }
+
         try {
             if (toConsole) {
-                writer = new BufferedWriter(new OutputStreamWriter(System.out));
+                writer = new BufferedWriter(new OutputStreamWriter(System.out, charset));
```

Comment:
Yes, but I was not sure that we could get the encoding buried in the writer. We need it to set the encoding attribute in the XML file. However, using the system properties `file.encoding` and `sun.stdout.encoding`, it's possible.